clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -576,7 +576,7 @@ public void handle(JoinGroupResponse joinResponse, RequestFuture<ByteBuffer> fut
             Errors error = joinResponse.error();
             if (error == Errors.NONE) {
                 if (isProtocolTypeInconsistent(joinResponse.data().protocolType())) {
-                    log.debug("JoinGroup failed due to inconsistent Protocol Type, received {} but expected {}",
+                    log.error("JoinGroup failed due to inconsistent Protocol Type, received {} but expected {}",
```

Comment:
Why are we logging this at error level when it's a client issue?

Reply:
@ijuma not sure I understand your question?

Reply:
Logging at error level should indicate a broker error. Bad client requests are a client error though and should not generate the logs at error level. Does that make sense?

Reply:
Atm we cannot say for sure, but it's likely that what's observed on KAFKA-9701 is a broker-side issue; we can either 1) add the check on broker-side across all members to make sure the selected protocol is consistent for everyone, so if the broker already made a wrong choice itself would log an ERROR, or 2) let it check on the client side. I think for trouble-shooting purposes option 2) is fine, and if we later discovered that this bug is actually on the client side I'm happy to revert this change after fixing it.

Reply:
So the `if (isProtocolTypeInconsistent(joinResponse.data().protocolType()))` check is like an assert for a bug in the broker implementation versus validating the join group request?